### PR TITLE
Module13

### PR DESCRIPTION
**Pull request by mpursell**
Module 13 PR - logging configured to write to local logs and to Loggly in JSON format.

loggly_screenshot_json_query
loggly_screenshot_PR

### CODE REVIEW

File: todo_app/app.py [hunk 1]
```diff
@@ -205,7 +208,7 @@ def authenticate():
 
         app_user = User(github_user["id"])
         app_user.role = user_db.get_user_role(userid=github_user["id"])
-        logging.info("user is authenticated as a: %s", app_user.role)
```
Comment (CorndelWithSoftwireDevOps) [inline todo_app/app.py]: In line with submission checklist, this is an opportune place for logging user id, as there isn't anywhere else you are doing it.

File: README.md [hunk 2]
```diff
@@ -303,4 +303,41 @@ Make sure the subscription id, resource group, storage and container strings are
...
+
+
+### Log Aggregation
+
+Log aggregation is provided by [Loggly](https://loggly.com).  You'll need to sign up for a Loggly account, and retrieve your customer token from Loggly. Setting up logging for Python apps in covered [here](https://documentation.solarwinds.com/en/success_center/loggly/content/admin/python-http.htm?cshid=loggly_python-http).
+
+You'll need a ```python.conf``` file in the root of the project:
+
+```python
+[handlers]
```
Comment (CorndelWithSoftwireDevOps) [inline README.md:320]: This approach is quite neat. You will need to add a way to configure at least the log level via environment variable though, as per submission checklist **The log level is configurable via an environment variable**

**Review by CorndelDevOps — CHANGES_REQUESTED** [review]
Good work getting logging with Loggly working and documented.

There is one small comment to add additional logging information and another one to allow log level to be controlled via environment variable. These should bring the PR in line with submission checklist.

Note that this PR contains code from previous modules, which hasn't been merged in yet. These were not reviewed in this PR. Make sure you don't merge it, in case there are changes required to previous modules' code

File: README.md (hunk 2 above)
Comment (mpursell) [reply README.md:320]: Added log level override via environment variable.

**Comment by JackMead** [comment]
Thanks for making the changes - I can see the commit where you added Log level override with an environment variable so happy to approve this, although it looks like you've removed it since. As it currently stands, I'm not sure how you're planning to get this working outside of a local environment however - you aren't able to check in the python.conf file, so how will you get that in place in production?

A standard pattern might be to inject the secret data via the environment variables, and then have your python.conf handle the non-sensitive aspects. Happy for you to merge when you're ready, but do consider how you might manage this process!

**Pull request merged**